src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs

```diff
@@ -639,9 +640,17 @@ public async Task Public_Interop_Upgrade_Success(string uri)
                 VersionPolicy = HttpVersionPolicy.RequestVersionOrLower
             })
             {
-                using HttpResponseMessage responseA = await client.SendAsync(requestA).WaitAsync(TimeSpan.FromSeconds(20));
-                Assert.Equal(HttpStatusCode.OK, responseA.StatusCode);
-                Assert.NotEqual(3, responseA.Version.Major);
+                try
+                {
+                    using HttpResponseMessage responseA = await client.SendAsync(requestA).WaitAsync(TimeSpan.FromSeconds(20));
+                    Assert.Equal(HttpStatusCode.OK, responseA.StatusCode);
+                    Assert.NotEqual(3, responseA.Version.Major);
+                }
+                catch (HttpRequestException ex) when (ex.InnerException is SocketException se && se.SocketErrorCode == SocketError.NetworkUnreachable)
+                {
+                    _output.WriteLine($"Unable to establish non-H/3 connection to {uri}: {ex}");
```

Comment:
Is the output written to stdout if the test completes successfully? I thought it happens only on errors.

Should we perhaps use SkipTestException?

Reply:
This is written to test output and stored in testResults.xml, you can see it even in success.